UnityProject/Assets/Scripts/UI/Systems/AdminTools/DevTools/GUI_DevTileChanger.cs

```diff
@@ -272,47 +295,70 @@ public void OnActionButtonClick(int buttonType)
 
 		#region Clicking
 
-		private void OnClick()
+		private bool OnClick()
 		{
-			if(currentAction == ActionType.None) return;
-
-			//Right click to stop placing
-			if (Input.GetMouseButtonDown(1))
-			{
-				currentAction = ActionType.None;
-				modeText.text = currentAction.ToString();
-				return;
-			}
-
-			//Ignore click if pointer is hovering over GUI
-			if (EventSystem.current.IsPointerOverGameObject())
-			{
-				return;
-			}
+			if(currentAction == ActionType.None) return true;
 
 			//Clicking once
-			if (Input.GetMouseButtonDown(0))
+			if (CommonInput.GetMouseButtonDown(0))
 			{
+				dragStartPos = MouseInputController.MouseWorldPosition.RoundToInt();
+
 				switch (currentAction)
 				{
 					case ActionType.Place:
 						//Also remove if shift is pressed when placing for quick remove
 						if (Input.GetKey(KeyCode.LeftShift) || Input.GetKey(KeyCode.RightShift))
 						{
 							RemoveTile();
-							return;
+							return true;
 						}
 
 						PlaceTile();
-						return;
+						return true;
 					case ActionType.Remove:
 						RemoveTile();
-						return;
+						return true;
 					default:
 						Logger.LogError($"Unknown case: {currentAction.ToString()} in switch!");
-						return;
+						return true;
 				}
 			}
+
+			return false;
+		}
+
+		private void OnDrag()
+		{
+			if(currentAction == ActionType.None) return;
+
+			//Being held down
+			if (CommonInput.GetMouseButton(0)) return;
+
+			if (CommonInput.GetMouseButtonUp(0) == false) return;
+
+			if(dragStartPos == MouseInputController.MouseWorldPosition.RoundToInt()) return;
+
+			//End drag
+			switch (currentAction)
+			{
+				case ActionType.Place:
+					//Also remove if shift is pressed when placing for quick remove
+					if (Input.GetKey(KeyCode.LeftShift) || Input.GetKey(KeyCode.RightShift))
```

Comment:
Would it be sensible to do the inverse if in remove mode?

Reply:
On the note of remove mode, I still find it funny how the remove tool for admins actually works correctly yet the unity editor version is still broke

Reply:
> Would it be sensible to do the inverse if in remove mode? Or do you see shift as to guarantee remove mode.
> 
> A little hint in the UI might be nice too if it doesn't have one already.

Have made it just do the reverse, not sure what UI thing i could add?

Reply:
Could replace the mode buttons with w/ a toggle group and change the current mode string to a hint label. Up to you though.